// next-superjson/dist/index.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.findPagesDir = exports.withSuperjson = void 0;
var fs_1 = __importDefault(require("fs"));
var path_1 = __importDefault(require("path"));
function withSuperjson(withSuperjson) {
    if (withSuperjson === void 0) { withSuperjson = {}; }
    return function (nextConfig) {
        if (nextConfig === void 0) { nextConfig = {}; }
        return __assign(__assign({}, nextConfig), { webpack: function (config, options) {
                var _a = withSuperjson;
                var isServer = options.isServer, dev = options.dev, dir = options.dir;
                var pagesDir = findPagesDir(dir);
                config.module = config.module || {};
                config.module.rules = config.module.rules || [];
                config.module.rules.push({
                    test: /\.(tsx|ts|js|mjs|jsx)$/,
                    include: [pagesDir],
                    use: [
                        options.defaultLoaders.babel,
                        {
                            loader: 'babel-loader',
                            options: {
                                sourceMaps: dev,
                                plugins: [
                                    [
                                        require.resolve('babel-plugin-superjson-next'),
                                        {},
                                    ],
                                    require.resolve('@babel/plugin-syntax-jsx'),
                                    [
                                        require.resolve('@babel/plugin-syntax-typescript'),
                                        { isTSX: true },
                                    ],
                                ],
                            },
                        },
                    ],
                });
                if (typeof nextConfig.webpack === 'function') {
                    return nextConfig.webpack(config, options);
                }
                else {
                    return config;
                }
            } });
    };
}
exports.withSuperjson = withSuperjson;
// taken from https://github.com/vercel/next.js/blob/v12.1.5/packages/next/lib/find-pages-dir.ts
function findPagesDir(dir) {
    // prioritize ./pages over ./src/pages
    var curDir = path_1.default.join(dir, 'pages');
    if (fs_1.default.existsSync(curDir))
        return curDir;
    curDir = path_1.default.join(dir, 'src/pages');
    if (fs_1.default.existsSync(curDir))
        return curDir;
    // Check one level up the tree to see if the pages directory might be there
    if (fs_1.default.existsSync(path_1.default.join(dir, '..', 'pages'))) {
        throw new Error('No `pages` directory found. Did you mean to run `next` in the parent (`../`) directory?');
    }
    throw new Error("Couldn't find a `pages` directory. Please create one under the project root");
}
exports.findPagesDir = findPagesDir;
